### PR TITLE
add runas as a global state variable

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
Adds runas to `__context__` for states to specify what the underlying `cmd` module should use.

A recommendation for where to put the documentation for this would be appreciated!

### What issues does this PR fix or reference?
#37835

### Tests written?

No

### CODE REVIEW

File: salt/modules/cmdmod.py [hunk 1]
```diff
@@ -52,6 +52,8 @@
 __proxyenabled__ = ['*']
 # Define the module's virtual name
 __virtualname__ = 'cmd'
+# Define blank context for importing for grains modules
+__context__ = {}
```
Comment (cachedout) [inline salt/modules/cmdmod.py]: This concerns me. I don't want an import of this module to overwrite the context if it's already been injected by the loader. Could we do this only if it hasn't already been set?

File: salt/modules/cmdmod.py [hunk 2]
```diff
@@ -289,6 +291,9 @@ def _run(cmd,
 
     log_callback = _check_cb(log_callback)
 
+    if runas is None:
+        runas = __context__.get('runas', None)
```
Comment (cachedout) [inline salt/modules/cmdmod.py]: The `None` here is a default and unnecessary, FWIW.

File: salt/state.py [hunk 3]
```diff
@@ -1648,6 +1649,8 @@ def call(self, low, chunks=None, running=None, retries=1):
         else:
             ret = {'result': False, 'name': low['name'], 'changes': {}}
 
+	self.state_con['runas'] = low.get('runas', None)
```
Comment (cachedout) [inline salt/state.py]: Is this the indentation you intended?

Comment (gtmanfred) [inline salt/state.py]: Woops, it was not, i really need to trash my vimrc and start over, because it keeps disabling expandtab for some reason.

**Comment by gtmanfred** [comment]
updated to only use `__context__` in `_run()` if `__context__` is in `locals()`

**Comment by gtmanfred** [comment]
Should I also go through and add deprecation warnings for oxygen for all the user: options in all the other states?

**Comment by cachedout** [comment]
Cool. I like this a lot more. Let's get this merged after the tests finish. Thanks, @gtmanfred

**Review by cachedout — APPROVED** [review]

**Comment by gtmanfred** [comment]
Let me get documentation in real quick, I am going to put it here

https://docs.saltstack.com/en/latest/ref/states/requisites.html

**Comment by cachedout** [comment]
Go Go Jenkins!

**Pull request merged**